feat(MDXComponents): migrate markdown package to @carbon-labs (#519)
carbon-design-system/carbon-labs: packages/react/src/components/MDXComponents/overview.mdx

> <Meta title="MDX Components/MDX Documentation Guidelines" />
> # MDX Documentation Guidelines  

Reviewer: ```suggestion
<Meta title="MDX Components/Guidelines" />
# Guidelines  
```
We don't need to duplicate "MDX Components" in this heading.

This document outlines the best practices and rules for writing MDX documentation 
for components under `@carbon-labs`. 

Following these guidelines will ensure all documented components have the same styles
as the ones found in the [Carbon Design System](carbondesignsystem.com) page.

> ## Directory Structure  

Reviewer: ```suggestion
## Meta component
```
We use sentence casing for headings, per https://carbondesignsystem.com/guidelines/content/writing-style/#capitalization

- All `.mdx` files **must** be placed under:  
  ```
  src/components/ComponentName/docs/

>   ```

Reviewer: I don't think this is true any more. For example, the example component uses a `__docs__` directory. We can remove this. Our MDX gets used if it's an MDX file, and it doesn't have an attached Component Story Format (CSF) like the autodocs MDX have.

> - The `Meta` component must be included at the top of the file with a `title` in the format:  

Reviewer: ```suggestion
- The `Meta` Storybook MDX component must be included at the top of the file with a `title`. This allows you to determine where the file is included in the side navigation, instead of it being placed based on directory path by default.  
```

  ```mdx
  <Meta title="ComponentName/Example" />
  ```

> ## Unsupported Features  

Reviewer: ```suggestion
## Unsupported features  
```

This library does **not** support the following MDX features:
- **Frontmatter** (metadata at the beginning of the file)
- **HTML comments** (e.g., `<!-- comment -->`)
- **Markdown image syntax** (e.g., `![alt text](./image.png)`)

> ## Using Images  

Reviewer: ```suggestion
## Using images  
```

Instead of markdown image syntax, import images and use the `<img />` tag:  
```mdx
import image from './images/1.png';

<img
  alt="example"
  src={image}
/>
```

> ## Available MDX Components  

Reviewer: Add the rest of the MDX components here too, like Preview.

> ## Available MDX Components  

Reviewer: ```suggestion
## Available MDX components  
```

The following Carbon MDX components are available for use in documentation:  

- `H1`, `H2`, `H3`, `H4`, `H5`, `H6`  
- `P`
- `OL` (Ordered List), `UL` (Unordered List), `LI` (List Item)  
- `Link`  
- `Blockquote`  
- `PageTable`  
- `Caption`  
- `Code`  
- `AnchorLinks`, `AnchorLink`  
- `Column`, `Row`, `Grid`  
- `DoDont`, `DoDontRow`  
- `GifPlayer`  
- `PageDescription`  
- `StorybookDemo`  
- `Tabs`, `Tab`  
- `InlineNotification`  
- `ArtDirection`  

> ## Unsupported MDX Components  

Reviewer: ```suggestion
## Unsupported MDX components  
```

> The following component is **not** supported in this documentation,
> but note that it is used in other Carbon sites:

Reviewer: ```suggestion
The following component are **not** currently supported in this documentation, although they may be available in Carbon websites:
```
